packages/vscode-extension/src/project/project.ts

```diff
@@ -163,14 +143,14 @@ export class Project
     );
 
     this.disposables.push(
-      workspace.onDidChangeConfiguration(async (event: ConfigurationChangeEvent) => {
+      workspace.onDidChangeConfiguration((event: ConfigurationChangeEvent) => {
         if (event.affectsConfiguration("launch")) {
           const config = getCurrentLaunchConfig();
-          const oldAppRoot = this.appRootFolder?.getAppRoot();
+          const oldAppRoot = this.appRootFolder;
           if (config.appRoot === oldAppRoot) {
             return;
           }
-          const newAppRoot = await this.setupAppRoot();
+          const newAppRoot = this.setupAppRoot();
```

Comment:
I think `setupAppRoot` shouldn't return anything -- I think it's a bit confusing when a function is used both for side effects (here, updating the object state) and returning a value.
Just call `setupAppRoot` here and use `this.appRootFolder` instead.